agent/consul/acl.go

```diff
@@ -1702,6 +1702,36 @@ func (f *aclFilter) filterServiceList(services *structs.ServiceList) {
 	*services = ret
 }
 
+// filterGatewayServices is used to filter gateway to service mappings based on ACL rules.
+func (f *aclFilter) filterGatewayServices(mappings *structs.GatewayServices) {
+	svcs := *mappings
+	var authzContext acl.AuthorizerContext
+
+	for i := 0; i < len(svcs); i++ {
+		s := svcs[i]
+
+		var drop bool
+		s.Gateway.FillAuthzContext(&authzContext)
+		if !f.allowService(s.Gateway.ID, &authzContext) {
+			drop = true
+			f.logger.Debug("dropping gateway from result due to ACLs", "gateway", s.Gateway.ID)
```

Comment:
Do you have to clarify what flavor of gateway you're talking about in the method name and this log message?

Reply:
I like that idea, just added that message here:
https://github.com/hashicorp/consul/blob/terminating-gw-disco/agent/consul/internal_endpoint.go#L315-L319